crates/cli-support/src/test_runner.rs

```diff
@@ -0,0 +1,627 @@
+            TestMode::Deno => b.deno(true)?,
+            TestMode::Browser { .. }
+            | TestMode::DedicatedWorker { .. }
+            | TestMode::SharedWorker { .. }
+            | TestMode::ServiceWorker { .. } => {
+                if test_mode.no_modules() {
+                    b.no_modules(true)?
+                } else {
+                    b.web(true)?
+                }
+            }
+        };
+
+        if self.split_linked_modules {
+            b.split_linked_modules(true);
+        }
+
+        let coverage = self.coverage_args(file_name);
+
+        // The debug here means adding some assertions and some error messages to the generated js
+        // code.
+        //
+        // It has nothing to do with Rust.
+        b.debug(self.debug)
+            .input_module(module, wasm)
+            .emit_start(false)
+            .generate(&tmpdir)
+            .context("executing `wasm-bindgen` over the Wasm file")?;
+        shell.clear();
+
+        match test_mode {
+            TestMode::Node { no_modules } => node::execute(
+                module,
+                &tmpdir,
+                tests,
+                !no_modules,
+                coverage,
+                self.nocapture,
+                self.include_ignored,
+                &self.node_bin,
+                &self.node_path,
+                &self.node_args,
+            )?,
+            TestMode::Deno => {
+                deno::execute(module, &tmpdir, tests, self.nocapture, self.include_ignored)?
+            }
+            TestMode::Browser { .. }
+            | TestMode::DedicatedWorker { .. }
+            | TestMode::SharedWorker { .. }
+            | TestMode::ServiceWorker { .. } => {
+                let headless = !self.no_headless;
+                let srv = server::spawn(
+                    &if headless {
+                        "127.0.0.1:0".parse().unwrap()
+                    } else if let Some(address) = &self.address {
+                        address.parse().unwrap()
+                    } else {
+                        "127.0.0.1:8000".parse().unwrap()
+                    },
+                    headless,
+                    module,
+                    &tmpdir,
+                    tests,
+                    test_mode,
+                    self.no_origin_isolation,
+                    coverage,
+                    self.nocapture,
+                    self.include_ignored,
+                )
+                .context("failed to spawn server")?;
+                let addr = srv.server_addr();
+
+                // TODO: eventually we should provide the ability to exit at some point
+                // (gracefully) here, but for now this just runs forever.
+                if !headless {
+                    println!("Interactive browsers tests are now available at http://{addr}");
+                    println!();
+                    println!("Note that interactive mode is enabled because `NO_HEADLESS`");
```

Comment:
Can you leave a `// TODO` here?

Reply:
What is the `TODO`? This was code that was copied:
https://github.com/wasm-bindgen/wasm-bindgen/blob/ba94c0dc77e6e6a911efe318d85d4df3f36c0cb2/crates/cli/src/bin/wasm-bindgen-test-runner/main.rs#L361-L372

Reply:
Are you saying to remove the prints in favor of a `TODO`?

Reply:
Done